tracks/r/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,57 @@
+### Reasonable solution
+
+This solution is easy to follow, has minimal re-assignments, and uses conditional logic in such a way to make the code more readable.
+
+```r
+bob <- function(input) {
+  trimmed_input <- trimws(input)
+
+  is_shouted_statement <- is_shouting(trimmed_input)
+  is_question_statement <- is_question(trimmed_input)
+
+  if (nchar(trimmed_input) == 0) {
+    "Fine. Be that way!"
+  } else if (is_shouted_statement && is_question_statement) {
+    "Calm down, I know what I'm doing!"
+  } else if (is_question_statement) {
+    "Sure."
+  } else if (is_shouted_statement) {
+    "Whoa, chill out!"
+  } else {
+    "Whatever."
+  }
+}
+
+is_question <- function(statement) { grepl("\\?$", statement) }
+is_shouting <- function(statement) { grepl("^[A-Z]+$", letters_only(statement)) }
+letters_only <- function(string) { gsub("[^a-zA-Z]", "", string) }
+```
+
+The learning goal here is to get comfortable with regular-expressions, pattern matching, and common string operations. Secondarily, having students check multiple conditions in an orderly fashion bolsters their conditional flow control.
+
+
+### Common suggestions
+
+- Readability can easily be degraded with dense conditional statements, or especially when trying to index on a vector using a Boolean vector (although decent examples exist, like the one below); have students try to make their answer as intelligible as possible.
+- If students are performing the same operations on `input` in their if statements, suggest that they create a variable instead, so the computer doesn't have to do more work than necessary.
+- Discuss how R returns values from a function if they exclusively use `return()` commands, or their solution could easily be altered to have no explicit return statements, like above.
```

Comment:
This should probably be:

- if they use explicit return commands, discuss how R returns values from a function and suggest they remove the explicit returns